### PR TITLE
New Error Reporter: Fix trailing whitespace

### LINKED ISSUE #7185: New error reporter produces trailing whitespace
```

Error: Expected primary expression.
 --> recovery_ast_constructor/input.sol:5:27: 
  |
5 |     balances[tx.origin] = ; // missing RHS.
  |                           ^

Warning: Recovered in Statement at ';'.
 --> recovery_ast_constructor/input.sol:5:27: 
  |
5 |     balances[tx.origin] = ; // missing RHS.
  |                           ^

```

### PR DESCRIPTION
**Pull request by Marenz**
fixes #7185

### CODE REVIEW

**Comment by chriseth** [comment]
Please remove the other commit.

File: liblangutil/SourceReferenceFormatterHuman.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ void SourceReferenceFormatterHuman::printSourceLocation(SourceReference const& _
 
 	// line 0: source name
 	frameColored() << string(leftpad, ' ') << "--> ";
-	m_stream << _ref.sourceName << ":" << (_ref.position.line + 1) << ":" << (_ref.position.column + 1) << ": " << '\n';
+	m_stream << _ref.sourceName << ":" << (_ref.position.line + 1) << ":" << (_ref.position.column + 1) << ":" << '\n';
```
Comment (bshastry) [inline liblangutil/SourceReferenceFormatterHuman.cpp:76]: Does it make sense to add a test case expects warning/error without trailing spaces?

**Review by bshastry — COMMENTED** [review]
Apart from the extra commit, LGTM. May be nice to add a test case that will flag a potential regression in the future

File: liblangutil/SourceReferenceFormatterHuman.cpp (hunk 1 above)
Comment (Marenz) [reply liblangutil/SourceReferenceFormatterHuman.cpp:76]: I'd added/adjusted a test case but the whole feature doesn't have a single test and I felt this little change didn't warrant writing one now.

**Comment by Marenz** [comment]
Removed extra commit

**Review by bshastry — APPROVED** [review]

**Pull request merged**